relay-sampling/src/lib.rs

```diff
@@ -84,1522 +84,39 @@ pub use dsc::DynamicSamplingContext;
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.environment", &["debug"], true),
-                    eq("trace.user.segment", &["paid", "vip", "free"], true),
-                ]),
-            ),
-            (
-                "multiple transactions",
-                and(vec![glob("trace.transaction", &["t22", "trans*", "t33"])]),
-            ),
-            (
-                "case insensitive user segments",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.environment", &["debug"], true),
-                    eq("trace.user.segment", &["ViP", "FrEe"], true),
-                ]),
-            ),
-            (
-                "multiple user environments",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq(
-                        "trace.environment",
-                        &["integration", "debug", "production"],
-                        true,
-                    ),
-                    eq("trace.user.segment", &["vip"], true),
-                ]),
-            ),
-            (
-                "case insensitive environments",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.environment", &["DeBuG", "PrOd"], true),
-                    eq("trace.user.segment", &["vip"], true),
-                ]),
-            ),
-            (
-                "all environments",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.user.segment", &["vip"], true),
-                ]),
-            ),
-            (
-                "undefined environments",
-                and(vec![
-                    glob("trace.release", &["1.1.1"]),
-                    eq("trace.user.segment", &["vip"], true),
-                ]),
-            ),
-            ("match no conditions", and(vec![])),
-        ];
-
-        let dsc = DynamicSamplingContext {
-            trace_id: Uuid::new_v4(),
-            public_key: ProjectKey::parse("abd0f232775f45feab79864e580d160b").unwrap(),
-            release: Some("1.1.1".into()),
-            user: TraceUserContext {
-                user_segment: "vip".into(),
-                user_id: "user-id".into(),
-            },
-            replay_id: Some(Uuid::new_v4()),
-            environment: Some("debug".into()),
-            transaction: Some("transaction1".into()),
-            sample_rate: None,
-            sampled: None,
-            other: BTreeMap::new(),
-        };
-
-        for (rule_test_name, condition) in conditions.iter() {
-            let failure_name = format!("Failed on test: '{rule_test_name}'!!!");
-            assert!(condition.matches(&dsc), "{failure_name}");
-        }
-    }
-
-    #[test]
-    /// test matching for various rules
-    fn test_matches_events() {
```

Comment:
replaced by `test_condition_matching` in `evaluation.rs`. 
 
I no longer test both dsc and event here as this crate is supposed to be agnostic to different types that implement `Getter`